docs/proposals/proposal-audit-log.md

```diff
@@ -0,0 +1,256 @@
+| PROJECT_SETTINGS_UPDATED       | Changes made to project **Acme API** settings by **admin@acme.com**.                                      |
+| ORGANIZATION_SETTINGS_UPDATED  | Changes made to organization **Acme** settings by **admin@acme.com**.                                     |
+| TARGET_SETTINGS_UPDATED        | Changes made to target **Acme dev API** settings by **admin@acme.com**.                                   |
+| SCHEMA_POLICY_SETTINGS_UPDATED | Changes made to schema policy settings under an org or a project by **admin@acme.com**.                   |
+
+
+## Audit log UI
+
+Download would be triggered by a `Download CSV` button in Hive UI.
+
+We would provide these filters:
+- **date range** - to avoid pulling entire history of events
+- **by user** - in case somebody wants to see actions made by a user
+- **by project** - to scope down the logs to a specific project (by default audit log contains data from all projects of the org)
+- **by target** - to scope down it even more
+- **by action** - type or group of action types
+
+
+### Graphql API for UI
+
+Graphql Schema to support a UI with these filters looks like this:
+
+```graphql
+
+  scalar Date
+  scalar JSON
+
+  enum AuditLogEventAction {
+    USER_INVITED
+    USER_JOINED
+    EXPIRED_INVITE_HIT
+    PROJECT_CREATED
+    TARGET_CREATED
+    ROLE_CREATED
+    ROLE_ASSIGNED
+    USER_REMOVED
+    ORG_TRANSFERRED
+    SCHEMA_CHECKED
+    SCHEMA_PUBLISH
+    SCHEMA_DELETED
+    PROJECT_SETTINGS_UPDATED
+    ORGANIZATION_SETTINGS_UPDATED
+    TARGET_SETTINGS_UPDATED
+    SCHEMA_POLICY_SETTINGS_UPDATED
+  }
+
+  type AuditLog {
+    event_time: Date!
+    user_id: String
+    user_email: String
+    organization_id: String
+    project_id: String
+    project_name: String
+    target_id: String
+    target_name: String
+    schema_version_id: String
+    event_action: AuditLogEventAction!
+    event_details: JSON
+  }
+
+  type AuditLogEdge {
+    cursor: String
+    node: AuditLog
+  }
+
+  type AuditLogConnection {
+    edges: [AuditLogEdge]
+    pageInfo: PageInfo
+  }
+
+  type PageInfo {
+    endCursor: String
+    hasNextPage: Boolean
+  }
+
+  input AuditLogFilter {
+    startDate: Date
+    endDate: Date
+    userId: String
+    userEmail: String
+    projectId: String
+    targetId: String
+    eventAction: String
+  }
+
+  type AuditLogFileExport {
+    url: String
+    filters: AuditLogFilter
+    validUntil: Date
+    createdAt: Date
+  }
+
+  type ExportResultEdge {
+    cursor: String
+    node: AuditLogFileExport
+  }
+
+  type ExportResultConnection {
+    edges: [ExportResultEdge]
+    pageInfo: PageInfo
+  }
+
+  type Query {
+    auditLogs(filter: AuditLogFilter, first: Int, after: String): AuditLogConnection
+    auditLogExports(first: Int, after: String ): ExportResultConnection
+  }
+
+  type Mutation {
+    exportAuditLogsToFile(filter: AuditLogFilter): AuditLogFileExport
```

Comment:
I guess here, we will show a link to the file, right? And filters, to render the thing correctly?

Reply:
Yes I think we want to give users ability to filter in exports too, not just in UI.
url is a field in the AuditLogFileExport, so FE can pick that when doing the mutation